src/EnergyPlus/CTElectricGenerator.cc

```diff
@@ -96,127 +96,59 @@ namespace CTElectricGenerator {
-        CalcCTGeneratorModel(GenNum, RunFlag, MyLoad, FirstHVACIteration);
-        UpdateCTGeneratorRecords(RunFlag, GenNum);
+        // If we didn't find it, fatal
+        ShowFatalError("LocalCombustionTurbineGeneratorFactory: Error getting inputs for combustion turbine generator named: " +
+                       objectName); // LCOV_EXCL_LINE
+        // Shut up the compiler
+        return nullptr; // LCOV_EXCL_LINE
     }
 
-    void SimCTPlantHeatRecovery(std::string const &EP_UNUSED(CompType), // unused1208
-                                std::string const &CompName,
-                                int const EP_UNUSED(CompTypeNum), // unused1208
-                                int &CompNum,
-                                bool const EP_UNUSED(RunFlag),
-                                bool &InitLoopEquip,
-                                Real64 &EP_UNUSED(MyLoad),
-                                Real64 &MaxCap,
-                                Real64 &MinCap,
-                                Real64 &OptCap,
-                                bool const EP_UNUSED(FirstHVACIteration) // TRUE if First iteration of simulation
-    )
+    void CTGeneratorData::simulate(const EnergyPlus::PlantLocation &EP_UNUSED(calledFromLocation),
+                                   bool EP_UNUSED(FirstHVACIteration),
+                                   Real64 &EP_UNUSED(CurLoad),
+                                   bool EP_UNUSED(RunFlag))
     {
-
         // SUBROUTINE INFORMATION:
-        //       AUTHOR         BGriffith
-        //       DATE WRITTEN   March 2008
+        //       AUTHOR         Dan Fisher
+        //       DATE WRITTEN   Sept. 2000
         //       MODIFIED       na
         //       RE-ENGINEERED  na
 
-        // PURPOSE OF THIS SUBROUTINE:
-        // Fill data needed in PlantLoopEquipments
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-
-        if (GetCTInput) {
-            GetCTGeneratorInput();
-            GetCTInput = false;
-        }
+        // PURPOSE OF THIS SUBROUTINE: This is the CT Generator driver.  It
+        // gets the input for the models, initializes simulation variables, call
+        // the appropriate model and sets up reporting variables.
 
-        if (InitLoopEquip) {
-            CompNum = UtilityRoutines::FindItemInList(CompName, CTGenerator);
-            if (CompNum == 0) {
-                ShowFatalError("SimCTPlantHeatRecovery: CT Generator Unit not found=" + CompName);
-                return;
-            }
-            MinCap = 0.0;
-            MaxCap = 0.0;
-            OptCap = 0.0;
-            return;
-        } // End Of InitLoopEquip
+        // empty function to emulate current behavior as of conversion to using the PlantComponent calling structure.
+        // calls from the plant side... do nothing.
+        // calls from the ElectricPowerServiceManger call the init and calculation worker functions directly.
```

Comment:
This does beg the question whether this component really ever should've been a part of the plant system if the sim call doesn't really do anything.  But that's not a question for right now, this is looking good.